lexer.go

```diff
@@ -399,6 +399,30 @@ func unescapeIdent(s string) string {
 	}
 }
 
+func hasEscapedChars(s string) bool {
+	i := 0
+	for i < len(s) {
+		r, size := utf8.DecodeRuneInString(s[i:])
+		if i == 0 && !isFirstIdentChar(r) || i > 0 && !isIdentChar(r) {
+			return true
+		}
+		i += size
+	}
+	return false
+}
+func ifEscapedCharsAppendQuotedIdent(dst []byte, s string) []byte {
+	if hasEscapedChars(s) {
+		dst = utf8.AppendRune(dst, '"')
+		for i := 0; i < len(s); {
+			r, size := utf8.DecodeRuneInString(s[i:])
+			dst = utf8.AppendRune(dst, r)
+			i += size
+		}
+		dst = utf8.AppendRune(dst, '"')
+		return dst
+	}
+	return appendEscapedIdent(dst, s)
+}
```

Comment:
wdyt about the following change:
```suggestion
func ifEscapedCharsAppendQuotedIdent(dst []byte, s string) []byte {
	if hasEscapedChars(s) {
		return appendQuotedIdent(dst, s)
	}
	return appendEscapedIdent(dst, s)
}
```

The new `appendQuotedIdent` could contain logic wrapped in `if` from your initial implementation.

Then, we can reuse `appendQuotedIdent` later - see further comments

Reply:
I like this refactor - will add.